tunnel/tunnel.go

```diff
@@ -821,7 +823,10 @@ func (s *Server) newClientSession(ctx context.Context, session *tpb.Session, add
 	t := Target{ID: session.TargetId, Type: session.TargetType}
 	// If client is requesting a remote target, only call the bridge register handler.
 	// We might extend it to allow calling the customized handler in the future.
-	if _, ok := s.rTargets[t]; ok {
+	s.tmu.RLock()
```

Comment:
thanks for catching this. we can probably call server.clientFromTarget and check if the returned value is empty?